src/Microsoft.ML.TorchSharp/Utils/FileUtils.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal static class FileUtils
+    {
+        private const int WriteBatchSize = 1024 * 1024;     // this size seems good on performance
+        private static readonly Type[] _validTypes = new Type[]
+        {
+            typeof(short),
+            typeof(ushort),
+            typeof(int),
+            typeof(uint),
+            typeof(long),
+            typeof(ulong),
+            typeof(float),
+            typeof(double),
+        };
+
+        public static string LoadFromFileOrDownloadFromWeb(string path, string fileName, Uri url)
+        {
+            var contents = "";
+            var filePath = Path.Combine(path, fileName);
+            if (!File.Exists(filePath))
+            {
+                try
+                {
+                    using var webClient = new WebClient();
+                    contents = webClient.DownloadString(url);
+                }
+                catch (WebException e)
+                {
+                    throw new WebException($"File {fileName} not found and cannot be downloaded from {url}.\n" +
+                                           $"Error message: {e.Message}");
```

Comment:
>) [](http://example.com/codeflow?start=72&length=1)

You may pass the original thrown exception here too.

Reply:
Added it in.